server/src/Irony.cpp

```diff
@@ -258,17 +259,30 @@ void Irony::complete(const std::string &file,
       unsigned annotationStart = 0;
       bool typedTextSet = false;
 
-      if (availability == CXAvailability_NotAccessible ||
-          availability == CXAvailability_NotAvailable) {
-        continue;
-      }
-
       typedtext.clear();
       brief.clear();
       resultType.clear();
       prototype.clear();
       postCompCar.clear();
       postCompCdr.clear();
+      inaccess.clear();
+
+      // progress even if candidate is `CXAvailability_NotAccessible' due to the
+      // current Clang bug which evaluates a protected member to be inaccessible
+      // to a derived class which has valid access.
+      //
+      // The rationale vs previous behaviour (which would not show an
+      // inaccessible candidate) is that the user's compiler catches erroneous
+      // access anyway. Since this is a completion engine, showing more is
+      // better than less.
+      //
+      // https://github.com/Sarcasm/irony-mode/issues/367
+      if (availability == CXAvailability_NotAvailable) {
+        continue;
+      } else if (availability == CXAvailability_NotAccessible) {
+        inaccess = "t";
```

Comment:
You should always set inaccess to a value, otherwise if an e=8th fields is added things will get confusing.